penguin/skills/discovery.py

```
@@ -0,0 +1,174 @@
+    current = config
+    for key in keys:
+        if isinstance(current, dict):
+            current = current.get(key, default)
+        else:
+            current = getattr(current, key, default)
+        if current is default:
+            return default
+    return current
+
+
+def get_skills_config(config: Any) -> Dict[str, Any]:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `config`

(ANN401)